src/badge.jsx

```diff
@@ -0,0 +1,111 @@
+const React = require('react');
+const Typography = require('./styles/typography');
+const DefaultRawTheme = require('./styles/raw-themes/light-raw-theme');
+const ThemeManager = require('./styles/theme-manager');
+const StylePropable = require('./mixins/style-propable');
+
+// Badge
+export default React.createClass({
+  displayName: 'Badge',
+  mixins: [StylePropable],
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  getChildContext () {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+  propTypes: {
+    className: React.PropTypes.string,
+    primary: React.PropTypes.bool,
+    secondary: React.PropTypes.bool,
+    labelColor: React.PropTypes.string,
+    backgroundColor: React.PropTypes.string,
+    style: React.PropTypes.object,
+    badgeStyle: React.PropTypes.object,
+    badgeContent: React.PropTypes.node.isRequired,
+  },
+  getInitialState() {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };
+  },
+  getDefaultProps() {
+    return {
+      className: '',
+      primary: false,
+      secondary: false,
+      labelColor: null,
+      backgroundColor: null,
+      style: {},
+      badgeStyle: {},
+    };
+  },
+  componentWillReceiveProps(nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({
+      muiTheme: newMuiTheme,
+    });
+  },
+  getStyles() {
+    const theme = this.state.muiTheme.badge;
+
+    const backgroundColor = this.props.backgroundColor
+      ? this.props.backgroundColor
+      : this.props.primary
+        ? theme.primaryColor
+        : this.props.secondary
+          ? theme.secondaryColor
+          : theme.color;
+
+    const labelColor = this.props.labelColor
+      ? this.props.labelColor
+        : this.props.primary
+          ? theme.primaryTextColor
+          : this.props.secondary
+            ? theme.secondaryTextColor
+            : theme.textColor;
+
+    return {
+      root: {
+        position: 'relative',
+        display: 'inline-block',
+        marginRight: 20,
+      },
+      badge: {
+        display: 'flex',
+        flexDirection: 'row',
+        flexWrap: 'wrap',
+        justifyContent: 'center',
+        alignContent: 'center',
+        alignItems: 'center',
+        position: 'absolute',
+        top: -11,
```

Comment:
I've added a variable 'radius' inside of the render function which is used to calculate top/right/width/height of badge and padding of root.  I've added padding, similar to IconButton, to contain the badge and help it's position in relation to elements around it.  I'll wait to squash commits until I get the OK on positioning.